Shared/sdk/SharedUtil.File.h

```diff
@@ -102,6 +102,8 @@ namespace SharedUtil
     WString FromUTF8(const SString& strPath);
     SString ToUTF8(const WString& strPath);
 
+    std::vector<std::string> ListDir(const char* szPath) noexcept;
```

Comment:
`noexcept` is dangerous here because `ListDir` method can throw an exception(bad_alloc for example).

Reply:
Bad_alloc can happen all the time, doesnt mean we should surround all mta code into multiple huge try-catch blocks. Other than that, no exceptions are thrown in ListDir function (except the possibility of an empty string but its easily solvable)

Reply:
`noexcept shall not be placed on a method that can even hypothetically throw an exception.

Reply:
Bad_alloc is the only exception here that can throw. All other are handled (fs functions throw under the fs_error class).
C functions dont throw at all